### PR TITLE
updating ruby and restforce with additional bug check

### PR DESCRIPTION
**Pull request by LoganArnett**
This is for ticket [#169188828](https://www.pivotaltracker.com/story/show/169188828)

### CODE REVIEW

File: app/javascript/utils/fieldSpecs.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const getRenderType = (value) => {
 }
 
 const dateIsJson = (date) => {
-  return (typeof date === 'object' && 'month' in date && 'day' in date && 'year' in date)
+  return (date && typeof date === 'object' && 'month' in date && 'day' in date && 'year' in date)
```
Comment (LoganArnett) [inline app/javascript/utils/fieldSpecs.js:19]: Found this to be breaking with certain date parsing

File: Gemfile [hunk 2]
```diff
@@ -84,7 +84,7 @@ gem "omniauth-salesforce", "~> 1.0"
 gem "omniauth-rails_csrf_protection"
 
 # Added at 2017-10-24 00:03:53 -0700 by dkaplan:
-gem "restforce", "~> 3.1"
+gem "restforce", "~> 4.2"
```
Comment (LoganArnett) [inline Gemfile:87]: Updated both of these and QA'd through the app to make sure general functionality is still as expected

**Comment by akegan** [comment]
please update the ruby version in the README too

**Review by akegan — APPROVED** [review]
Changes look good, trying to test locally to see if the restforce update helped at all, but that shouldn't block merge.

File: README.md [hunk 3]
```diff
@@ -10,8 +10,8 @@ Only showing rspec tests for now:
 Cross-browser testing done with <a href="https://www.browserstack.com/"><img src="./Browserstack-logo@2x.png?raw=true" height="36" ></a>
 
 ## Setup
-* Use Ruby 2.3.7 (Set the version using [RVM](https://rvm.io/rvm/install) or [rbenv](https://github.com/rbenv/rbenv))
-* Install [Bundler](https://github.com/bundler/bundler) for this version of Ruby `gem install bundler -v 1.16.2`
+* Use Ruby 2.6.3 (Set the version using [RVM](https://rvm.io/rvm/install) or [rbenv](https://github.com/rbenv/rbenv))
```
Comment (LoganArnett) [inline README.md:13]: I updated Ruby version and bundler version here, is there anywhere else in the README?

Comment (LoganArnett) [reply README.md:13]: @akegan

Comment (akegan) [reply README.md:13]: oh whoops. i hadn't seen that update, looks good then!

**Pull request merged**